### PR TITLE
Remove Nextcloud 25 support

### PR DESCRIPTION
**Pull request by mejo-**
### 📝 Summary

* [x] Removed Nextcloud 25 support
* [x] Removed CI test runs against stable25 
* [x] Removed special handling of Cypress tests for Nextcloud 25
* [x] Removed split of `ACLStorageWrapper` into `ACLStorageWrapper25` and `ACLStorageWrapper26`
* [x] Removed 25-only `PageReferenceProvider`
* [x] Removed PHP7.4 support

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] Tests (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [x] Documentation is not required

### CODE REVIEW

File: rector.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+return function (RectorConfig $rectorConfig) {
+	$rectorConfig->paths([
+		__DIR__ . '/lib',
+		__DIR__ . '/tests',
+	]);
+	$rectorConfig->phpVersion(PhpVersion::PHP_80);
+	$rectorConfig->importNames();
+	$rectorConfig->indent('	', 1);
+	$rectorConfig->sets([
+		SetList::PHP_74,
```
Comment (blizzz) [inline rector.php:19]: :shrug:

**Review by juliushaertl — APPROVED** [review]

**Review by blizzz — APPROVED** [review]
It's a loooong PR.

I left you some nitpicks you might wanna double check. May include false positives.

File: cypress/e2e/pages.spec.js [hunk 2]
```diff
@@ -316,15 +311,15 @@ describe('Page', function() {
 
 	describe('Using the search providers to search for a page', function() {
 		it('Search for page title', function() {
-			cy.get('.unified-search a, .unified-search button, button.unified-search__button').click()
+			cy.get('.unified-search a, button.unified-search__button, .unified-search-menu button').click()
```
[the remark this answers is not in the document]
Comment (mejo-) [reply cypress/e2e/pages.spec.js:314]: That fixes the selector to make it work on master/stable29 again. Not exactly related to this PR but a by-catch.

File: cypress/e2e/pages.spec.js [hunk 3]
```diff
@@ -316,15 +311,15 @@ describe('Page', function() {
...
 			cy.get('.unified-search__form input, .unified-search-modal input')
 				.type('Day')
 			cy.get('.unified-search__results-collectives-pages, .unified-search-modal__results')
 				.should('contain', 'Day 1')
 		})
 
 		it('Search for page content', function() {
-			cy.get('.unified-search a, .unified-search button, button.unified-search__button').click()
+			cy.get('.unified-search a, button.unified-search__button, .unified-search-menu button').click()
```
[the remark this answers is not in the document]
Comment (mejo-) [reply cypress/e2e/pages.spec.js:322]: Same here.

File: rector.php (hunk 1 above)
Comment (mejo-) [reply rector.php:19]: I wanted to check if it found old leftovers :laughing:

**Pull request merged**